packages/flutter_tools/test/general.shard/build_system/targets/macos_test.dart

```diff
@@ -223,21 +285,50 @@ void main() {
       copyFrameworkCommand,
       lipoInfoFatCommand,
       lipoVerifyX86_64Command,
-      FakeCommand(command: <String>[
-          'lipo',
-          '-output',
-          binary.path,
-          '-extract',
-          'x86_64',
-          binary.path,
-      ]),
+      lipoExtractX86_64Command,
     ]);
 
     await const DebugUnpackMacOS().build(environment);
 
     expect(processManager, hasNoRemainingExpectations);
   });
 
+  testUsingContext('Fails if framework dSYM found within framework but copy fails', () async {
```

Comment:
Can you also add tests here
https://github.com/flutter/flutter/blob/aa934ac119d007ad2e9dac7dbd75e41c1ee17ac8/packages/flutter_tools/test/host_cross_arch.shard/macos_content_validation_test.dart#L140-L152
and here
https://github.com/flutter/flutter/blob/aa934ac119d007ad2e9dac7dbd75e41c1ee17ac8/packages/flutter_tools/test/host_cross_arch.shard/ios_content_validation_test.dart#L165
(not those exact lines, but you can see how the App.framework.dSYM is being checked)